docs/dev_guides/changing_splink/testing.md

```diff
@@ -0,0 +1,283 @@
+---
+tags:
+  - Testing
+  - Pytest
+  - Backends
+---
+# Testing in Splink
+
+Tests in Splink make use of the [pytest](https://docs.pytest.org) framework. You can find them in [the tests folder](https://github.com/moj-analytical-services/splink/tree/master/tests).
+
+Splink tests can be broadly categorised into three sets:
+
+* 'Core' tests - these are tests which test some specific bit of functionality which does not depend on any specific SQL dialect. They are usually unit tests - examples are testing [`InputColumn`](https://github.com/moj-analytical-services/splink/blob/master/tests/test_input_column.py) and testing the [latitude-longitude distance calculation](https://github.com/moj-analytical-services/splink/blob/master/tests/test_lat_long_distance.py).
+* Tests for specific backends - these are tests which run against a specific SQL backend, and test some feature peculiar to this backend. There are not many of these, as Splink is designed to run very similarly independent of the backend used.
+* Backend-agnostic tests - these are tests which run against some SQL backend, but which are written in such a way that they can run against many backends by making use of the [backend-agnostic testing framework](#backend-agnostic-testing). The majority of tests are of this type.
+
```

Comment:
Do we need a caveat here/somewhere else to explain why Athena isn't included as a backend to test? I.e. because of needing a connection to an AWS account.